### DISCUSSION TITLE
How to use repository action variables (plain text) on workflow action?

### DISCUSSION THREAD

**Discussion by pedrojfds** [community/community#42133] [Actions] [answered]
### Select Topic Area

Product Feedback

### Body

I've tried to use action variables defined on github /settings/variables/actions in my workflow but the value is never found.
It works with secrets, but the values we want to keep do not need to be secrets. 

These were my tests, both as environment variables and repository variables:
Tried to read them as `${{ env.TEST }}`, `${{ secrets.TEST }}` and `$TEST` and nothing worked.

Workflow:

![image](https://user-images.githubusercontent.com/112859896/208460452-4497c160-3d87-4f07-b953-3bbe968122a3.png)

Output:

![image](https://user-images.githubusercontent.com/112859896/208460503-c92d40ee-1157-4e84-a7fb-973902133b1d.png)

Configuration on /settings/variables/actions :

![image](https://user-images.githubusercontent.com/112859896/208460538-389f461b-5f5b-49a9-a9af-92ac7028e386.png)

- How can I read those variables on YML workflow?

**Comment by mbaum0** [comment, marked as answer]
It is in the `vars` context. So you should be able to access the repository variables by `${{ vars.TEST2 }}`

**Reply by savoygrizzly** [reply]
I can't believe I had to dig this deep to find this here. @github your doc sucks.

**Reply by vaimdev** [reply]
Agree, I find this here also !

**Reply by Silverlan** [reply]
Same here. Unfortunate that this hasn't improved in almost two years.

**Reply by ghost** [reply]
pretty awful docs alright. We have Enterprise support. even if you tell them not to they just pump every query thru 'AI' and give you a response like you're some kind of idiot. takes a lot of effort to even get GitHub support to escalate Enterprise requests to a competent human.

**Reply by sourcecodemage** [reply]
I want to the thank the OP for asking this question and for mbaum0 for answering it. I read over github's "using variables" page and it doesn't explain this. It said to us ${} for vars used in the action , and ${{}} for actions that the runner needs before the step is run. eg a "when" needs ${{}}. a "run:" would need ${}.  "secrets." was there, "vars." was not.

**Reply by FrancoDominguez** [reply]
This gives me a "Context access might be invalid: IMAGE_NAME" error, does anybody know why? Nothing wrong with variables defined within the file, same for secrets, I'm really just having trouble with action variables ${{ vars.IMAGE_NAME }}

**Reply by filipopo** [reply]
Just came here too for this same problem...

**Reply by nicholaswmin** [reply]
> pretty awful docs alright. We have Enterprise support. even if you tell them not to they just pump every query thru 'AI' and give you a response like you're some kind of idiot. takes a lot of effort to even get GitHub support to escalate Enterprise requests to a competent human.

Id like ro thank this guy for stating the obvious

**Reply by raghav-phonebox** [reply]
Thank you, the deployment finally works.

**Comment by anandrathnas** [comment]
@Github Please put a note on [/{orgOrUser}/{repo}/settings/variables/actions](https://github.com/octocat/Hello-World/settings/variables/actions) page itself as a quick reference. Googling mostly takes us to https://docs.github.com/en/actions/learn-github-actions/variables this page is a TON of TMI and actually its kinda unrelated too

**Comment by kingthorin** [comment]
Is it possible for a workflow to write to or store a value in a repo’s `vars` context?

**Reply by mbaum0** [reply]
You could use the GitHub api.
https://docs.github.com/en/rest/actions/variables?apiVersion=2022-11-28#update-a-repository-variable

See [using github cli in workflows](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/using-github-cli-in-workflows)

**Comment by mdesco-crgl** [comment]
Does anybody know if `${{ vars. }} works in composite actions?

```
    - name: Unit test with pytest
      shell: bash
      env:
        SOURCE_DIR: ${{ vars.SOURCE_DIR }}
      run: pytest ${SOURCE_DIR} -m "${{ inputs.markers }}" ${{ inputs.pytest_args}}
```
results in 
```
Error: crgl-ca/hn-fabfile/ci-access/.github/actions/ci/action.yml (Line: 39, Col: 21): Unrecognized named-value: 'vars'. Located at position 1 within expression: vars.SOURCE_DIR
Error: GitHub.DistributedTask.ObjectTemplating.TemplateValidationException: The template is not valid. crgl-ca/hn-fabfile/ci-access/.github/actions/ci/action.yml (Line: 39, Col: 21): Unrecognized named-value: 'vars'. Located at position 1 within expression: vars.SOURCE_DIR
```

doesn't work as direct reference, in runner env, inputs, etc. anywhere.
